### PR TITLE
Design for the add assignment madlib form

### PR DESCRIPTION
**Pull request by mborsare**
This PR makes the add a class form match the styles established in the
add a class/outcome form.

add assignment

https://trello.com/c/QW12ljja

### CODE REVIEW

File: app/views/manage_assessments/assignments/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+  <%= form.input :problem,
+      label: t(".problem"),
+      placeholder: "Problem 4"%>
+
   <%= form.submit t(".add") %>
+  <%= link_to t("helpers.cancel_button"), :back %>
 <% end %>
+
+<script type = "text/javascript">
+  $(function() {
```
Comment (mborsare) [inline app/views/manage_assessments/assignments/new.html.erb]: This is straight from stackoverflow. Is there a better way to do it?

File: app/views/manage_assessments/assignments/new.html.erb [hunk 2]
```diff
@@ -1,5 +1,45 @@
-<%= simple_form_for @assignment, url: manage_assessments_outcome_coverage_assignments_path do |form| %>
-  <%= form.input :name, label: t(".assignment") %>
-  <%= form.input :problem, label: t(".problem") %>
+<%= content_for :full_bleed do %>
```
Comment (mborsare) [inline app/views/manage_assessments/assignments/new.html.erb]: Should we turn this into a partial since we use it here and on the add a class/outcome form?

File: app/views/manage_assessments/assignments/new.html.erb [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+          $(this).addClass("is-empty");
+          $(this).removeClass("has-items");
+        } else {
+          $(this).removeClass("is-empty");
+          $(this).addClass("has-items");
+        }
+      });
+    }
+
+    inp.blur(err);
```
Comment (derekprior) [inline app/views/manage_assessments/assignments/new.html.erb]: Do you want to do this only on blur, or would you like it done any time the user enters text (don't wait until they blur)?

File: app/views/manage_assessments/assignments/new.html.erb [hunk 4]
```diff
@@ -1,5 +1,45 @@
-<%= simple_form_for @assignment, url: manage_assessments_outcome_coverage_assignments_path do |form| %>
-  <%= form.input :name, label: t(".assignment") %>
-  <%= form.input :problem, label: t(".problem") %>
+<%= content_for :full_bleed do %>
+  <div class="madlib-header">
+    <h1 class="headline-narrow">
+      Add assignment
+    </h1>
+    <h2 class="headline-narrow-subhead">
+      <%= @assignment.outcome_coverage.coverage.course.number %>
```
Comment (margaretoti) [inline app/views/manage_assessments/assignments/new.html.erb]: This can be made `assignment.outcome.course.name`. @derekprior, I am playing around in the console in an attempt to make this less brittle. Any ideas? I am happy to add some code in the model to accomplish this.

File: app/views/manage_assessments/assignments/new.html.erb [hunk 5]
```diff
@@ -1,5 +1,45 @@
...
+      &middot;
+      <%= @assignment.outcome_coverage.coverage.course.name %>
```
Comment (margaretoti) [inline app/views/manage_assessments/assignments/new.html.erb]: `assignment.outcome.course.name`

File: app/views/manage_assessments/assignments/new.html.erb (hunk 4 above)
Comment (derekprior) [inline app/views/manage_assessments/assignments/new.html.erb]: We would need a string of associations and delegates such that `assignment.course_number` works. 

something like:

```ruby
class Assignment
  has_one :coverage, through: :outcome_coverage
  delegate :course_number, to: :coverage
end

class Coverage
  delegate :number, to: :course, prefix: true
end
```

... should work.

File: app/views/manage_assessments/assignments/new.html.erb (hunk 1 above)
Comment (derekprior) [inline app/views/manage_assessments/assignments/new.html.erb]: This is pretty close to how I'd do it, but we can clean it up a bit.

File: app/views/manage_assessments/assignments/new.html.erb (hunk 3 above)
Comment (mborsare) [inline app/views/manage_assessments/assignments/new.html.erb]: Keeping it on blur is fine – unless there's some sort of performance issue.

File: app/views/manage_assessments/assignments/new.html.erb [hunk 6]
```diff
@@ -1,5 +1,45 @@
...
+
   <%= form.submit t(".add") %>
+  <%= link_to t("helpers.cancel_button"), :back %>
 <% end %>
+
+<script type = "text/javascript">
+  $(function() {
+    var inp = $("input[type='text']");
+
+    function checkInput() {
```
Comment (mborsare) [inline app/views/manage_assessments/assignments/new.html.erb]: @derekprior I renamed this function, what do you think?

**Comment by derekprior** [comment]
@mborsare I added commit with some refactoring of the JS, Ruby, and i18n. This looks good to me.

I didn't extract a template for the full bleed yet. Partials that accept blocks are always a little weird and hard to follow. A helper might make more sense. I might play with a bit in a different PR.
